### PR TITLE
Mobile 4550

### PR DESCRIPTION
**Pull request by dpalou**

### CODE REVIEW

File: src/addons/mod/quiz/components/index/addon-mod-quiz-index.html [hunk 1]
```diff
@@ -45,64 +45,74 @@
...
-                                <p *ngFor="let sentence of attempt.readableState">{{ sentence }}</p>
-                            </ion-col>
-                            <ion-col class="ion-text-center ion-hide-md-down" *ngIf="quiz.showMarkColumn" role="cell">
-                                <p>{{ attempt.readableMark }}</p>
-                            </ion-col>
-                            <ion-col class="ion-text-center" *ngIf="quiz.showGradeColumn" role="cell">
-                                <p>{{ attempt.readableGrade }}</p>
-                            </ion-col>
-                        </ion-row>
+                        {{ 'addon.mod_quiz.attempt' | translate:{ $a: attempt.attempt } }}
```
Comment (crazyserver) [inline src/addons/mod/quiz/components/index/addon-mod-quiz-index.html:111]: I'd probably use some heading here, h3 probably.

Comment (crazyserver) [reply src/addons/mod/quiz/components/index/addon-mod-quiz-index.html:111]: You can also use role="heading" aria-level="3" on ion-label to maintain aspect.

File: src/addons/mod/quiz/components/index/index.scss [hunk 2]
```diff
@@ -1,33 +1,43 @@
...
 
-    .addon-mod_quiz-table {
-        ion-card-content {
-            padding-left: 0;
-            padding-right: 0;
-        }
+    .addon-mod_quiz-attempt-title-info {
+        text-align: end;
+        padding-top: 8px;
+        padding-bottom: 8px;
```
Comment (crazyserver) [inline src/addons/mod/quiz/components/index/index.scss]: It's not mandatory yet, but we should start using spacing variables from design system on all paddings and margins. Those variables should not be overriden (probably in the future we can override them form a "density setting".

File: src/addons/mod/quiz/components/index/index.scss [hunk 3]
```diff
@@ -1,33 +1,43 @@
...
+        visibility: hidden;
+    }
+
+    hr {
+        background-color: var(--stroke);
+        height: 1px;
+        margin: 0px 16px;
+    }
+
+    ion-accordion:nth-child(odd) {
```
Comment (crazyserver) [inline src/addons/mod/quiz/components/index/index.scss]: Is this "zebra design" something coming from UX? I think that if only one section is displayed at a time, we don't need it. However the opened section can be highlighted somehow. Very minor but if we can find the chance to discuss it, better :-)

File: src/addons/mod/quiz/services/quiz.ts [hunk 4]
```diff
@@ -412,73 +411,59 @@ export class AddonModQuizProvider {
...
-    getAttemptReadableStateName(state: string): string {
+    getAttemptStateColor(state: string, finishedOffline = false): string {
+        if (finishedOffline) {
+            return 'medium';
+        }
+
         switch (state) {
             case AddonModQuizAttemptStates.IN_PROGRESS:
-                return Translate.instant('addon.mod_quiz.stateinprogress');
+                return 'warning';
```
Comment (crazyserver) [inline src/addons/mod/quiz/services/quiz.ts:457]: We could use CoreIonicColorNames here

File: src/addons/mod/quiz/components/index/index.ts [hunk 5]
```diff
@@ -409,20 +403,19 @@ export class AddonModQuizIndexComponent extends CoreCourseModuleMainActivityComp
         this.checkCompletion();
 
         // Verify that user can see the review.
-        const attemptId = this.autoReview.attemptId;
+        const attempt = attempts.find(attempt => attempt.id == this.autoReview?.attemptId);
```
Comment (crazyserver) [inline src/addons/mod/quiz/components/index/index.ts:393]: Better ===

**Review by crazyserver — CHANGES_REQUESTED** [review]

File: src/addons/mod/quiz/components/index/addon-mod-quiz-index.html (hunk 1 above)
Comment (dpalou) [reply src/addons/mod/quiz/components/index/addon-mod-quiz-index.html]: Using h3.

File: src/addons/mod/quiz/components/index/index.scss (hunk 2 above)
Comment (dpalou) [reply src/addons/mod/quiz/components/index/index.scss]: Done!

File: src/addons/mod/quiz/components/index/index.scss (hunk 3 above)
Comment (dpalou) [reply src/addons/mod/quiz/components/index/index.scss]: It was in the UX design, yes, so I left it as it is.

File: src/addons/mod/quiz/services/quiz.ts (hunk 4 above)
Comment (dpalou) [reply src/addons/mod/quiz/services/quiz.ts]: Done, nice catch!

**Review by crazyserver — DISMISSED** [review]

**Review by crazyserver — APPROVED** [review]

**Pull request merged**